tools/lock_inspect/step1.rb

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env ruby
+
+require 'pg'
+require 'yaml'
+require 'optparse'
+
+class ApplicationNameIncompleteError < RuntimeError
+end
+
+def error_exit(e, conn, error_message = '', exit_code = 1)
+  puts e.message
+  puts error_message unless error_message.empty?
+  conn.close if conn
+  exit(exit_code)
+end
+
+def generate_default_options
+  {
+    :output               => "evm_connection.yml",
+    :server               => "127.0.0.1",
+    :port                 => 5432,
+    :user                 => nil,
+    :password             => nil,
+    :ignore_parsing_error => nil
+  }
+end
+
+def parse_commandline_options
+  options = generate_default_options
+  op = OptionParser.new do |opts|
+    opts.banner = <<-BANNER
+Usage: step1.rb [options] [output_file | #{options[:output]}]
+Output pg_stat_activity in YAML format.
+    BANNER
+
+    opts.separator ""
+    opts.separator "Specific options:"
+
+    opts.on("-h", "--host [STRING]",
+            "The postgre server address, default to #{options[:server]}") do |host|
+      options[:host] = host
+    end
+
+    opts.on("", "--port [INTEGER]",
+            "The postgre server port, default to #{options[:port]}") do |port|
+      options[:port] = port
+    end
+
+    opts.on("-u", "--user [STRING]",
+            "The user to login to postgre server, default to current user") do |user|
+      options[:user] = user
+    end
+
+    opts.on("-p", "--password [STRING]",
```

Comment:
Yes. This can be done in a more secure way since we only ask custom to run this script once dead lock happens. No need to run it periodically, so we can ask user to input password interactively.